conda/petsc/build.sh

```diff
@@ -54,6 +54,7 @@ BUILD_CONFIG=`cat <<"EOF"
   --download-superlu_dist=1 \
   --with-fortran-bindings=0 \
   --with-sowing=0 \
+  --with-64-bit-indices \
```

Comment:
@loganharbour I could see both sides for moose-petsc (conda). 

Use 64-bit will have better consistency for users.

Use 32-bit will save memory a small laptop calculations. I do not think we can save a lot.

Right now, we use 64-bit for conda

Reply:
Let us take the third opinion from @lindsayad

Reply:
Hm.... I'm not sure if I agree with this. Why add it now if it was doing fine before? We don't suggest conda to be used across nodes, and single machines won't have enough memory to even support 64 bit...

Reply:
I don't think that it matters all too much. People using conda often don't really know what's going on. Going with 32 bit probably wouldn't save a lot of memory as @fdkong said...if you're running a big enough problem your heap profiling would probably show other bottlenecks...and of course as @loganharbour you can't actually run a big problem on a single machine! And at the same time you can't really profile with conda anyways unless you want to see a bunch of garbage from both PETSc and libMesh. So I see all of this as a wash. I don't think the user is going to notice one way or another.

Reply:
Okay. I'm fine with this - if they're gonna run something big enough locally to run out of memory, it's not gonna matter what size the dof indices are

Reply:
If I had to lean one way or another...I would vote 64-bit conda. Based off the `update_and_rebuild_petsc.sh` script we are moving towards 64-bit being our canonical build. It makes sense to me to make sure our canonical build works with conda.

Reply:
OK, let us use 64-bit for conda. 

I was thinking a bit more. Even use conda, sometimes I still need to build my own PETSc, in such case, the inconsistency may cause headache when debugging because you see different things.